libraries/ms-common/src/stm32f0xx/spi.c

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include "spi.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "gpio.h"
+#include "status.h"
+#include "stm32f0xx.h"
+
+static SPI_TypeDef *spi_periph_map[] = { SPI1, SPI2 };
+
+static GPIOAddress s_cs_pin_map[] = { 0, 0 };
+
+static bool prv_is_periph_valid(const SPIPeriph spi_x) {
+  return spi_x == 0 || spi_x == 1;
+}
+
+static bool prv_are_settings_valid(const SPISettings *settings) {
+  return !(settings->polarity > SPI_CPOL_HIGH ||
+           settings->phase > SPI_CPHA_2EDGE ||
+           settings->baud_rate > SPI_BAUDRATE_256 ||
+           settings->first_bit > SPI_FIRSTBIT_LSB);
+}
+
+static bool prv_is_state_valid(const GPIOState state) {
+  return state < NUM_GPIO_STATE;
+}
+
+static uint8_t prv_exchange(SPIPeriph spi_x, uint8_t data) {
+  while (!SPI_I2SGetFlagStatus(SPI_I2S_FLAG_TXE)) {}
```

Comment:
You should be able to replace the `{}` with `;`, although I'm not sure what our stance is on that.
@karlding @ckitagawa